scripts/db-heal-test.py

```diff
@@ -0,0 +1,231 @@
+
+    # Export genesis to restore LiveDB
+    genesis = Path(tmp_path) / 'test_genesis.dat'
+
+    print(f"Restoration complete. Exporting LiveDB genesis block {latest_checkpoint}.")
+    result = subprocess.run(
+        ['go', 'run', './database/mpt/tool', 'export', '--block', str(latest_checkpoint), str(archive), str(genesis)],
+        stdout=c,
+        stderr=c,
+        cwd=carmen_root)
+    if result.returncode != 0:
+        # Print carmen logs if interation fails
+        text = c.read()
+        print(text)
+        sys.exit(1)
+
+    # Restore LiveDB
+    print("Export complete. Applying LiveDB genesis.")
+    try:
+        shutil.rmtree(live)
+        print("Live directory removed successfully.")
+    except FileNotFoundError:
+        print(f"Directory {live} does not exist.")
+        exit(1)
+    except PermissionError:
+        print(f"Permission denied to remove {live}.")
+        exit(1)
+    except Exception as e:
+        print(f"An error occurred: {e}")
+        exit(1)
+
+    print("Importing LiveDB genesis.")
+    result = subprocess.run(
+        ['go', 'run', './database/mpt/tool', 'import-live-db', str(genesis), str(live)],
+        stdout=c,
+        stderr=c,
+        cwd=carmen_root)
+    if result.returncode != 0:
+        # Print carmen logs if interation fails
+        text = c.read()
+        print(text)
+        sys.exit(1)
+
+    print(f"Iteration {i}/{number_of_iterations}")
+    # We restored to block X, although we need to start the app at +1 block because X is already done
+    first_block = latest_checkpoint + 1
+
+    print("Syncing restarted...")
+    command = [
+        './build/aida-vm-sdb', 'substate', '--validate',
+        '--db-tmp', tmp_path, '--carmen-schema', '5', '--db-impl', 'carmen',
+        '--aida-db', aida_db_path, '--no-heartbeat-logging', '--track-progress',
+        '--archive', '--archive-variant', 's5', '--archive-query-rate', '200',
+        '--carmen-checkpoint-interval', str(checkpoint_granularity_flag), '--db-src',
+        str(working_dir), '--skip-priming', '--tracker-granularity',
+        str(checkpoint_granularity_flag), str(first_block), str(last_block)
+    ]
+
+    os.chdir(aida_path)
+    with open(aida_log_file, 'w') as f:
+        process = subprocess.Popen(command, stdout=f, stderr=subprocess.STDOUT)
+    os.chdir(current_dir)
+
+    # Start monitoring the log file
+    latest_checkpoint = terminate_process_after(window, latest_checkpoint)
+
+    # Wait for the command to complete
+    process.wait()
+
+    if latest_checkpoint == -1:
+        sys.exit(1)
+
+    if last_working_dir:
+        print(f"Removing previous database {last_working_dir}")
+        shutil.rmtree(last_working_dir, ignore_errors=True)
+
+    genesis.unlink(missing_ok=True)
+# Clear anything leftover
+
+print(f"Clearing last database {working_dir} and log files.")
+aida_log_file.unlink(missing_ok=True)
+carmen_log_file.unlink(missing_ok=True)
+shutil.rmtree(working_dir, ignore_errors=True)
+
```

Comment:
nit: It is always nice to have a final message like "All tests passed" or "Success", just to know that the process just not crashed with something unforeseeable.